astroid/nodes/scoped_nodes.py

```diff
@@ -637,12 +651,27 @@ def fully_defined(self):
         """
         return self.file is not None and self.file.endswith(".py")
 
-    def statement(self):
+    @overload
+    def statement(self, future: Literal[None] = ...) -> "Module":
+        ...
+
+    @overload
+    def statement(self, future: Literal[True]) -> NoReturn:
+        ...
+
+    def statement(
+        self, future: Union[Literal[True]] = None
```

Comment:
```suggestion
        self, future: Optional[Literal[True]] = None
```

Reply:
Oops, I replaced the `|` from your suggestion. Should have payed more attention.

Reply:
I should have used the correct syntax for `astroid`.